warden/x/warden/keeper/rules.go

```diff
@@ -19,114 +19,240 @@ func (k Keeper) RegisterRules(reg *acttypes.RulesRegistry) {
+	}
+
+	return approveRule, rejectRule, nil
 }
 
-func (k Keeper) newKeyRequestRule(ctx context.Context, msg *v1beta3.MsgNewKeyRequest) (acttypes.Rule, error) {
+func (k Keeper) newKeyRequestRule(ctx context.Context, msg *v1beta3.MsgNewKeyRequest) (acttypes.Rule, acttypes.Rule, error) {
 	space, err := k.SpacesKeeper.Get(ctx, msg.SpaceId)
 	if err != nil {
-		return acttypes.Rule{}, err
+		return acttypes.Rule{}, acttypes.Rule{}, err
 	}
 
-	if space.SignRuleId > 0 {
-		return k.actKeeper.GetRule(ctx, space.SignRuleId)
-	} else {
-		return space.RuleNewKeyRequest(), nil
+	approveRule, err := k.getApproveNewKeyRequestRule(ctx, space)
+
+	if err != nil {
+		return acttypes.Rule{}, acttypes.Rule{}, err
+	}
+
+	rejectRule, err := k.getRejectNewKeyRequestRule(ctx, space)
+
+	if err != nil {
+		return acttypes.Rule{}, acttypes.Rule{}, err
 	}
+
+	return approveRule, rejectRule, nil
 }
 
-func (k Keeper) newSignRequestRule(ctx context.Context, msg *v1beta3.MsgNewSignRequest) (context.Context, acttypes.Rule, error) {
+func (k Keeper) newSignRequestRule(ctx context.Context, msg *v1beta3.MsgNewSignRequest) (context.Context, acttypes.Rule, acttypes.Rule, error) {
 	key, err := k.KeysKeeper.Get(ctx, msg.KeyId)
 	if err != nil {
-		return nil, acttypes.Rule{}, err
+		return nil, acttypes.Rule{}, acttypes.Rule{}, err
 	}
 
 	space, err := k.SpacesKeeper.Get(ctx, key.SpaceId)
 	if err != nil {
-		return nil, acttypes.Rule{}, err
+		return nil, acttypes.Rule{}, acttypes.Rule{}, err
 	}
 
-	rule, err := k.getKeyRule(ctx, space, key)
+	approveRule, err := k.getApproveKeyRule(ctx, space, key)
 	if err != nil {
-		return nil, acttypes.Rule{}, err
+		return nil, acttypes.Rule{}, acttypes.Rule{}, err
+	}
+
+	rejectRule, err := k.getRejectKeyRule(ctx, space, key)
+	if err != nil {
+		return nil, acttypes.Rule{}, acttypes.Rule{}, err
 	}
 
 	creator := k.actKeeper.GetActionCreator(ctx)
 
 	ctxWithVals, dataForSigning, err := k.executeAnalyzers(ctx, creator, msg.Analyzers, msg.Input)
 	if err != nil {
-		return nil, acttypes.Rule{}, errors.Wrapf(v1beta3.ErrAnalyzer, "%v", err)
+		return nil, acttypes.Rule{}, acttypes.Rule{}, errors.Wrapf(v1beta3.ErrAnalyzer, "%v", err)
 	}
 
 	if dataForSigning != nil {
 		msg.Input = dataForSigning
 	}
 
-	return ctxWithVals, rule, nil
+	return ctxWithVals, approveRule, rejectRule, nil
 }
 
-func (k Keeper) updateKeyRule(ctx context.Context, msg *v1beta3.MsgUpdateKey) (acttypes.Rule, error) {
+func (k Keeper) updateKeyRule(ctx context.Context, msg *v1beta3.MsgUpdateKey) (acttypes.Rule, acttypes.Rule, error) {
 	key, err := k.KeysKeeper.Get(ctx, msg.KeyId)
 	if err != nil {
-		return acttypes.Rule{}, err
+		return acttypes.Rule{}, acttypes.Rule{}, err
 	}
 
 	space, err := k.SpacesKeeper.Get(ctx, key.SpaceId)
 	if err != nil {
-		return acttypes.Rule{}, err
+		return acttypes.Rule{}, acttypes.Rule{}, err
 	}
 
-	rule, err := k.getKeyRule(ctx, space, key)
+	approveRule, err := k.getApproveUpdateKeyRule(ctx, space, key)
```

Comment:
It was `rule, err := k.getKeyRule(ctx, space, key)`, I think you don't need `getApproveUpdateKeyRule`. You need to replace this by  `getApproveKeyRule`

Reply:
I think we accidentally didn't use updateKeyRule

Reply:
The difference with getApproveKeyRule/getRejectKeyRule in returning RuleApproveUpdateKey/RuleRejectUpdateKey in last else branch. RuleUpdateKey not used before changes, I think it's a mistake.

Reply:
@Pitasi we need your opinion

Reply:
where is `getApproveKeyRule` defined?

Reply:
it was renamed to getApproveSignRule